lib/addons/karpenter/index.ts

```diff
@@ -332,46 +474,121 @@ export class KarpenterAddOn extends HelmAddOn {
 
         karpenterChart.node.addDependency(ns);
 
-        // (Optional) default provisioner
-        if ((Object.keys(subnetTags).length > 0) && (Object.keys(sgTags).length > 0)){
-            const provisioner = cluster.addManifest('default-provisioner', {
-                apiVersion: 'karpenter.sh/v1alpha5',
-                kind: 'Provisioner',
-                metadata: { name: 'default' },
-                spec: {
-                    providerRef: {
-                        name: "default"
+        // Deploy Provisioner (Alpha) or NodePool (Beta) CRD based on the Karpenter Version
+        if (this.options.NodePoolSpec){
```

Comment:
code style: I observe cases when the template is quite different but injected values are the same. Such code blocks with if/then take up space and make code script style that is tough to navigate. 
Consider using kubectlProvider.addManifest which allows passing replacement values to the templates. example [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/cloudwatch-adot-addon/index.ts#L100) and similar in the codebase.

In that style you can load two templates and pass the same set of values to get the right manifest. 

Can be deferred.

Reply:
will defer.